src/asm/utils/16_read_disk.asm

```diff
@@ -31,15 +31,20 @@ maybe_retry:
 	jnz again
 	jmp disk_error
 done:
+	lea si, DISK_SUCCESS_MESSAGE
+	call print_string_16
 	ret
-
 # Print the return code of interrupt
 disk_error:
-	lea bx, DISK_ERROR_MESSAGE
-	call print_string
-disk_error_loop:
-	jmp disk_error_loop
+	lea si, DISK_ERROR_MESSAGE
+	call print_string_16
+	xor dx, dx
+	mov dl, ah
```

Comment:
What guarantees that ah is not overriden by print_string_16?

Reply:
fixed, print_string saves state